src/entity/static_mesh.cpp

```diff
@@ -82,30 +77,17 @@ StaticMesh::StaticMesh(
         float_colors.push_back(vector_color);
     }
 
-    assert(glGetError() == GL_NO_ERROR);
     LOG_DEBUG(logging::opengl_logger, "float_colors {}", float_colors);
-    //LOG_DEBUG(logging::opengl_logger, "color texture id before {}", color_texture_);
     glGenTextures(1, &color_texture_);
-    assert(glGetError() == GL_NO_ERROR);
-    //LOG_DEBUG(logging::opengl_logger, "color texture id after binding {}", color_texture_);
     glBindTexture(GL_TEXTURE_1D, color_texture_);
-    assert(glGetError() == GL_NO_ERROR);
     glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
-    assert(glGetError() == GL_NO_ERROR);
-    // set the texture wrapping/filtering options (on the currently bound texture object)
     glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
-    assert(glGetError() == GL_NO_ERROR);
     glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
-    assert(glGetError() == GL_NO_ERROR);
     glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
-    assert(glGetError() == GL_NO_ERROR);
     glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-    assert(glGetError() == GL_NO_ERROR);
     // load and generate the texture
     glTexImage1D(GL_TEXTURE_1D, 0, GL_RGBA32F, float_colors.size(), 0, GL_RGBA, GL_FLOAT, float_colors.data());
-    assert(glGetError() == GL_NO_ERROR);
```

Comment:
NOOOOOOOO WHY DID YOU TAKE AWAY THE ASSERTS

You should replace them with LOG_CRITICALs because you cannot continue with an error